From 47000808e3b4ad84f4eddb649739717e86363f1a Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Fri, 6 Oct 2023 10:00:51 +0200
Subject: [PATCH] webossystem_injection.cc: fix some PalmSystem functions

* Disable length check for AddBannerMessage arguments in order to allow legacy banners to work.
* use correct signature for simulateMouseClick

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Pending

 .../injection/renderer/webossystem/webossystem_injection.cc | 6 ++----
 .../injection/renderer/webossystem/webossystem_injection.h  | 4 ++--
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/neva/injection/renderer/webossystem/webossystem_injection.cc b/src/neva/injection/renderer/webossystem/webossystem_injection.cc
index 8d6bfbd698..5b5bf27576 100644
--- a/src/neva/injection/renderer/webossystem/webossystem_injection.cc
+++ b/src/neva/injection/renderer/webossystem/webossystem_injection.cc
@@ -416,8 +416,8 @@ void WebOSSystemInjection::UseSimulatedMouseClicks(bool param) {
 }
 
 void WebOSSystemInjection::SimulateMouseClick(
-    const std::string& result0, const std::string& result1, bool result2) {
-  std::vector<std::string> arguments = { result0, result1 };
+    int pageX, int pageY, bool result2) {
+  std::vector<std::string> arguments = { std::to_string(pageX), std::to_string(pageY) };
   arguments.push_back(result2 ? "true" : "false");
   SendCommand("simulateMouseClick", arguments);
 }
@@ -432,8 +432,6 @@ void WebOSSystemInjection::RemoveBannerMessage(const std::string& param) {
 }
 
 void WebOSSystemInjection::AddBannerMessage(gin::Arguments* args) {
-  if (args->Length() < 5)
-    return;
 
   std::vector<std::string> arguments;
   auto context = args->GetHolderCreationContext();
diff --git a/src/neva/injection/renderer/webossystem/webossystem_injection.h b/src/neva/injection/renderer/webossystem/webossystem_injection.h
index 1dfb643f8d..a019b985df 100644
--- a/src/neva/injection/renderer/webossystem/webossystem_injection.h
+++ b/src/neva/injection/renderer/webossystem/webossystem_injection.h
@@ -107,8 +107,8 @@ class WebOSSystemInjection : public gin::Wrappable<WebOSSystemInjection>,
   void AddBannerMessage(gin::Arguments* args);
   void RemoveBannerMessage(const std::string& param);
   void ClearBannerMessages();
-  void SimulateMouseClick(const std::string& result0,
-                          const std::string& result1,
+  void SimulateMouseClick(int pageX,
+                          int pageY,
                           bool result2);
   void UseSimulatedMouseClicks(bool param);
   void Paste();
